CAIPs/caip-25.md

```diff
@@ -63,9 +93,9 @@ Example:
 {
   "id": 1,
   "jsonrpc": "2.0",
-  "method": "provider_authorization",
+  "method": "provider_authorize",
   "params": {
```

Comment:
shouldn't params be an array of two arrays containing various scope objects?

```typescript
type Scope = {
methods: string[];
events: string[];
}
type ChainScope = Scope;
type NamespaceScope = Scope & { chains: Array<CAIP2Id>; }
type RequiredScope = NamespaceScope | ChainScope;
type OptionalScope = RequiredScope;

type CAIP25Params = [NonEmptyArray<RequiredScope>, Array<OptionalScope>]
```

Reply:
Ooh, so the params is an object in the examples and an array in the text? I hesitate to fix that either way by myself, let me check out if JSON-RPC syntax will make my job easier by disqualifying one of the two options 😅

Reply:
https://www.jsonrpc.org/specification#parameter_structures
^ Seems it can be array ("by-position") OR object ("by-name"), and "by-name" sounds more appropriate to this kind of finicky, versioned, open-world interoperability use-case... by-position sounds too brittle and easy to confuse or break.

Reply:
so i'm personally leaning towards keep object and make the text make the example :D

Reply:
haha yes, I would also prefer object-based -- well that solves that :)